test/functional/diff/screen_spec.lua

```diff
@@ -0,0 +1,515 @@
+local lfs = require('lfs')
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+
+local eq = helpers.eq
+local feed = helpers.feed
+local eval = helpers.eval
+local clear = helpers.clear
+local funcs = helpers.funcs
+local nvim_prog = helpers.nvim_prog
+local write_file = helpers.write_file
+
+local function nvim_prog_abs()
+  -- system(['build/bin/nvim']) does not work for whatever reason. It needs to
+  -- either be executable searched in $PATH or something starting with / or ./.
+  if nvim_prog:match('[/\\]') then
+    return funcs.fnamemodify(nvim_prog, ':p')
+  else
+    return nvim_prog
+  end
+end
+
+describe('Diff mode screen', function()
+  describe('generic', function()
+    local fname = 'Xtest-functional-diff-screen-1'
+    local fname_2 = fname .. '.2'
+
+    local withScreen = function(w, h, action)
+      screen = Screen.new(w, h)
+      screen:attach()
+      action(screen)
+      screen:detach()
+    end
+
+    local reread = function()
+      feed(':e<cr><c-w>w:e<cr><c-w>w')
+    end
+
+    local verifyFillerAndInternal = function(action)
+      for _, cmd in ipairs({':set diffopt=filler<cr>', ':set diffopt+=internal<cr>'}) do
+        feed(cmd)
+        reread()
+        action()
+      end
+    end
+
+    setup(function()
+      clear()
+      os.remove(fname)
+      os.remove(fname_2)
+      feed('<c-w>o:enew!<cr>')
+      feed(':e ' .. fname_2 .. ' | diffthis<cr>')
+      feed(':vnew ' .. fname .. ' | diffthis<cr>')
+    end)
+
+    teardown(function()
+      os.remove(fname)
+      os.remove(fname_2)
+      feed('<c-w>o:enew!<cr>')
+    end)
+
+    it('Add a line in beginning of file 2', function()
+      if helpers.pending_win32(pending) then return end
+      write_file(fname, "1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n", false)
+      write_file(fname_2, "0\n1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n", false)
+      withScreen(40, 16, function(screen)
+        verifyFillerAndInternal(function()
+          screen:sleep(1)  -- resolve "screen not changed"
```

Comment:
`screen:sleep(1)` doesn't resolve "screen not changed" neither semantically (nor technically I thought, if it does consistenty, `screen.lua` has a bug).

It is better to turn this inside out: invoke`it()` for each alternative in the loop, and redo the entire test from scratch.

Reply:
Thanks, I was just going to change these tests to explicitly check the two screens.